lib/bookingsync_application/controllers/bookingsync_universe_api_access.rb

```diff
@@ -0,0 +1,78 @@
+module BookingsyncApplication
+  module Controllers
+    module BookingsyncUniverseApiAccess
+      private
+
+      def authenticate_account!
+        if enable_for_bookingsync_universe_api? && auth = request.headers["Authorization"].presence
+          response = Faraday.new(url: bookingsync_url).get do |req|
+            req.url oauth_path
+            req.headers["Authorization"] = auth
+          end
+          if response.success?
+            @scope = AuthorizationScope.from_response(response)
+
+            session[:account_id] = scope.account_id
+          else
+            render json: response.body, status: response.status and return
+          end
+        else
+          super
+        end
+      end
+
+      def bookingsync_url
+        "#{ENV['BOOKINGSYNC_URL']}"
+      end
+
+      def oauth_path
+        "/api/v3/oauth"
+      end
+
+      def enable_for_bookingsync_universe_api?
+        false
+      end
+
+      def scope
+        @scope
+      end
+
+      def booingsync_universe_authorize_request!(*required_scopes)
+        if !scope.allows_access_for?(Array(required_scopes).map(&:to_s))
+          render json: { errors: [ { code: :forbidden } ] }, status: 403 and return
```

Comment:
I think it's the same as on core, no? It should be 403 error as the request is in fact authenticated, so this cannot be 401. Depending on the scopes it can be either 20x or 403.